Multicluster tutorial (#681)
kiali/kiali.io: content/en/docs/Tutorials/multicluster/08-Install-Istio-west-cluster.md

---
title: "Install Istio on West cluster"
description: "Install Istio on the remote cluster"
weight: 8
---

> This installation will be different as this cluster will be a remote:

Reviewer: Maybe it is possible to explain a bit why is different? Like the control plane will not be installed, just the CRDs, and it will be managed by a external CP?

```
kubectl create namespace istio-system --context $CLUSTER_WEST

kubectl create secret generic cacerts -n istio-system --context $CLUSTER_WEST \
      --from-file=certs/$CLUSTER_WEST/ca-cert.pem \
      --from-file=certs/$CLUSTER_WEST/ca-key.pem \
      --from-file=certs/$CLUSTER_WEST/root-cert.pem \
      --from-file=certs/$CLUSTER_WEST/cert-chain.pem

kubectl --context=$CLUSTER_WEST annotate namespace istio-system topology.istio.io/controlPlaneClusters=$CLUSTER_EAST
kubectl --context=$CLUSTER_WEST label namespace istio-system topology.istio.io/network=network2

cat <<EOF > $CLUSTER_WEST.yaml
apiVersion: install.istio.io/v1alpha1
kind: IstioOperator
spec:
  profile: remote
  values:
    istiodRemote:
      injectionPath: /inject/cluster/$CLUSTER_WEST/net/network2
    global:
      remotePilotAddress: ${DISCOVERY_ADDRESS}
EOF

istioctl install -y --context=$CLUSTER_WEST -f $CLUSTER_WEST.yaml
```

We will also install a Prometheus instance on the remote. We will federate both Prometheus, with the east's one being the place where all metrics will be gathered together:

```

> kubectl apply -f https://raw.githubusercontent.com/istio/istio/release-1.18/samples/addons/prometheus.yaml --context $CLUSTER_WEST

Reviewer 2: Use the Prometheus addon that comes with the Istio the reader is using:

```suggestion
kubectl apply -f $ISTIO_DIR/samples/addons/prometheus.yaml --context $CLUSTER_WEST
```

```

An important step is to create a secret on east cluster allowing it to fetch information of the remote cluster:

```
istioctl x create-remote-secret \
    --context=$CLUSTER_WEST \
    --name=$CLUSTER_WEST | \
    kubectl apply -f - --context=$CLUSTER_EAST
```

Finally, we create the east-west gateway

```

> samples/multicluster/gen-eastwest-gateway.sh \

Reviewer: Shouldn't be? 
`$ISTIO_DIR/samples/...`

    --mesh mesh1 --cluster $CLUSTER_WEST --network network2 | \
    istioctl --context=$CLUSTER_WEST install -y -f -

```

## Prometheus federation

> An important design decision for Kiali was to decide that it will continue consuming data from one Prometheus instance per all clusters. For this reason, Prometheus needs to be federated, meaning that all the remote’s metrics should be fetched by the main Prometheus.

Reviewer 3: ```suggestion
Even in a multi-cluster environment, Kiali expects unified metrics from a single Prometheus endpoint for all clusters. In this tutorial, we will federate the two Prometheus instances, meaning that all the remote’s metrics should be fetched by the main Prometheus.
```

> An important design decision for Kiali was to decide that it will continue consuming one Prometheus instance per all clusters. For this reason, Prometheus needs to be federated, meaning that all the remote’s metrics should be fetched by the main Prometheus.

Reviewer 2: ```suggestion
An important design decision for Kiali was to decide that it will continue consuming data from one Prometheus instance per all clusters. For this reason, Prometheus needs to be federated, meaning that all the remote’s metrics should be fetched by the main Prometheus.
```
